core-bundle/src/Resources/contao/classes/DataContainer.php

```diff
@@ -858,6 +860,33 @@ protected function switchToEdit($id)
 		return $strUrl . (!empty($arrKeys) ? '&' : '') . (Input::get('table') ? 'table=' . Input::get('table') . '&amp;' : '') . 'act=edit&amp;id=' . rawurlencode($id);
 	}
 
+	/**
+	 * @throws AccessDeniedException
+	 */
+	protected function denyAccessUnlessGranted($attributes, $subject): void
+	{
+		/** @var Security $security */
+		$security = System::getContainer()->get('security.helper');
+
+		if ($security->isGranted($attributes, $subject))
+		{
+			return;
+		}
+
+		$message = 'Access denied.';
+
+		if ($subject instanceof DataContainerSubject)
+		{
+			$message = sprintf('Access denied to %s [%s].', $subject, $attributes);
```

Comment:
wouldn't this generate `Access denied to XXX [Array()].`?

Reply:
`$attributes` is always a string in our cases. Maybe rename to `$attribute` as they did in Symfony? https://github.com/symfony/symfony/commit/91249ed0bfac5609e6bac4e4613b388256aba9cc

Reply:
aah, I was confusing this with `$attributes` of the `DataContainerSubject` 🙃